### PR TITLE
MutableCubicSpline1D

### PR DESCRIPTION
**Pull request by AdamNorberg**
A demonstration of sorts of the intended use pattern for MutableCurve, this backs it up with CubicSpline1D. Adds a wrapper to implement ICurve<Real> over CubicSpline1D, which works with float instead (for now).

### CODE REVIEW

**Comment by AdamNorberg** [comment]
Note that this is branched off of my MutableCurve branch, so its history shows the commits in that branch as well.

File: engine/geometry/MutableCubicSpline1D.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using System.Diagnostics;
+using System.Linq;
+using FreedomOfFormFoundation.AnatomyEngine.Calculus;
+
+namespace FreedomOfFormFoundation.AnatomyEngine.Geometry
+{
+    /// <summary>
+    /// MutableCubicSpline1D uses MutableCurve to implement moveable point behavior for CubicSpline1D.
+    /// </summary>
+    public class MutableCubicSpline1D : MutableCurve<Mutable2DPoint, Real>
```
Comment (Lathreas) [inline engine/geometry/MutableCubicSpline1D.cs:27]: Since this is 1D, I'd again suggest moving this to the `Calculus` namespace.

**Review by Lathreas — COMMENTED** [review]
Apart from what I commented in the previous PR, LGTM so far!

**Comment by AdamNorberg** [comment]
I'm coming up pretty blank on effective ways to unit test this that aren't already covered by the MutablePiecewiseInterpolatedCurve patch. Suggestions welcome.

**Review by zennithn — APPROVED** [review]
LGTM. Will let you know if I discover any problems!

**Pull request merged**